contracts/DomainController.sol

```diff
@@ -0,0 +1,108 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title DomainController
+ * @dev A simple domain registry contract that stores domain records associated with Ethereum addresses of the controllers.
+ * Each controller can create multiple domains. Domains are reserved forever.
+ */
+contract DomainController {
+    address public owner;
+    uint256 public domainRegistrationFee;
+
+    struct Domain {
+        address controller;
+        uint256 registrationDate;
+    }
+
+    mapping(string => Domain) public domains;
+    string[] public registeredDomains;
+    mapping(address => string[]) public controllerDomains;
+
+    event DomainRegistered(string domain, address indexed controller, uint256 registrationDate);
+
+    /**
+     * @dev Constructor sets the contract owner and the initial domain registration fee.
+     */
+    constructor() {
+        owner = msg.sender;
+        domainRegistrationFee = 1 ether; // Fixed domain registration fee
+    }
+
+    /**
+     * @dev Modifier to check if the caller is the contract owner.
+     */
+    modifier onlyOwner() {
+        require(msg.sender == owner, "Only contract owner can call this function");
+        _;
+    }
+
+    /**
+     * @dev Allows the contract owner to set the domain registration fee.
+     * @param newFee The new domain registration fee.
+     */
+    function setDomainRegistrationFee(uint256 newFee) public onlyOwner {
```

Comment:
// line 44 and other function declared as public 
// it is better to use external for better gas optimization
// - 2 point